src/Orangebeard.Client/V3/ClientUtils/Logging/LogManager.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+
+namespace Orangebeard.Client.V3.ClientUtils.Logging
+{
+    /// <summary>
+    /// Class to manage all internal loggers.
+    /// </summary>
+    public class LogManager
+    {
+        static LogManager()
+        {
+        }
+
+        private static readonly Lazy<LogManager> LogManagerInstance = new Lazy<LogManager>(() => new LogManager());
+
+        /// <summary>
+        /// Returns single instance of <see cref="LogManager"/>
+        /// </summary>
+        public static LogManager Instance => LogManagerInstance.Value;
+
+        private string _baseDir = Environment.CurrentDirectory;
+
+
+        /// <summary>
+        /// Fluently sets BaseDir.
+        /// </summary>
+        /// <param name="baseDir"></param>
+        /// <returns></returns>
+        public LogManager WithBaseDir(string baseDir)
+        {
+            _baseDir = baseDir;
+
+            return this;
+        }
+
+        private static readonly object LockObj = new object();
+
+        private static Dictionary<Type, ILogger> _loggers;
+
+        /// <summary>
+        /// Gets or creates new logger for requested type.
+        /// </summary>
+        /// <param name="type">Type where logger should be registered for</param>
+        /// <returns><see cref="ILogger"/> instance for logging internal messages</returns>
+        private ILogger GetLogger(Type type)
+        {
+            if (type == null) throw new ArgumentNullException(nameof(type));
+
+            if (_loggers == null)
```

Comment:
Is it an option to put the assignment in the declaration - i.e. in line 41, `private static Dictionary<Type, ILogger> _loggers; = new Dictionary<Type, ILogger>()'`  ?

Reply:
agree. refactoring